src/dbsql.py

```diff
@@ -87,3 +91,96 @@ def build_parameter(params: dict) -> Parameter:
                 if p['name'] in query.params
             ]
         ).as_dict()
+
+    def create_alert(self, alert: Alert, target_folder: str, destination_id: str | None = None, warehouse_id: str | None = None) -> str:
+        """
+        Given a Redash alert, creates a Databricks alert
+
+        :param alert: Redash alert model
+        :param target_folder: target folder to create the alert in
+        :param destination_id: optional ID of the destination if schedule is set
+        :param warehouse_id: optional ID of the SQL warehouse to refresh query
+        """
+
+        target_folder_path = "folders/{0}".format(self._get_path_object_id(target_folder))
+
+        # First migrate the query
+        query_id = self.create_query(alert.query, target_folder_path)
+        result = self._create_alert_api_call(query_id, alert, target_folder_path)
+        if alert.schedule and destination_id and warehouse_id:
+            self._create_alert_schedule_api_call(result.id, alert.schedule, destination_id, warehouse_id)
+        return result.id
+
+    def _create_alert_api_call(self, query_id: str, alert: Alert, parent_folder: str):
+        """
+        Creates an alert in Databricks
+        """
+        return self.client.alerts.create(
+            name=alert.name,
+            options=AlertOptions.from_dict(alert.options),
+            query_id=query_id,
+            parent=parent_folder,
+            rearm=alert.rearm,
+        )
+
+    def _create_alert_schedule_api_call(self, alert_id: str, schedule: dict, destination_id: str, warehouse_id: str):
+        """
+        Creates an alert schedule in Databricks
+        """
+        return self.client.jobs.create(
+            name=f"Alert {alert_id} schedule",
```

Comment:
It will be useful to include the name of the alert that this schedule belongs to (not just the ID). wdyt?

Reply:
Added additional details to the description and change the title to have the name of the alert